python/ray/serve/_private/http_util.py

```diff
@@ -72,6 +73,14 @@ def __init__(self, content=None, status_code=200):
         self.status_code = status_code
         self.raw_headers = []
 
+        # Set request id.
+        self.raw_headers.append(
+            [
+                RAY_SERVE_REQUEST_ID,
+                ray.serve.context._serve_request_context.get().request_id,
+            ]
+        )
```

Comment:
Is it possible to do this only in the HTTP proxy instead of modifying the three separate codepaths?

Reply:
I will rebase after your pr landed. and restructure the code. :)

Reply:
Thank you 🙏